mlir/lib/IR/AffineMap.cpp

```diff
@@ -57,13 +60,34 @@ class AffineExprConstantFolder {
           expr, [](int64_t lhs, int64_t rhs) { return lhs * rhs; });
     case AffineExprKind::Mod:
       return constantFoldBinExpr(
-          expr, [](int64_t lhs, int64_t rhs) { return mod(lhs, rhs); });
+          expr,
+          [expr, this](int64_t lhs, int64_t rhs) -> std::optional<int64_t> {
+            if (rhs < 1) {
+              hasPoison_ = true;
+              return std::nullopt;
+            }
+            return mod(lhs, rhs);
+          });
     case AffineExprKind::FloorDiv:
       return constantFoldBinExpr(
-          expr, [](int64_t lhs, int64_t rhs) { return floorDiv(lhs, rhs); });
+          expr,
+          [expr, this](int64_t lhs, int64_t rhs) -> std::optional<int64_t> {
+            if (0 == rhs) {
```

Comment:
```suggestion
            if (rhs == 0) {
```

Reply:
Is it more reasonable to place constants on the left?

Reply:
> 

Is it more reasonable to place constants on the left?

Reply:
It looks unusual. Why would you do that? If that's to guard against accidental assignment, the warnings enabled by default in llvm already enforce that assignments are wrapped with an extra layer of parens.

Reply:
Thanks.